runtime/sema/type.go

```diff
@@ -4957,34 +4958,31 @@ func (t *CompositeType) MemberMap() *StringMemberOrderedMap {
 }
 
 func (t *CompositeType) SupportedEntitlements() (set *EntitlementOrderedSet) {
-	supportedEntitlements := t.supportedEntitlements
-	if supportedEntitlements != nil {
-		return supportedEntitlements
-	}
-
-	set = orderedmap.New[EntitlementOrderedSet](t.Members.Len())
-	t.Members.Foreach(func(_ string, member *Member) {
-		switch access := member.Access.(type) {
-		case *EntitlementMapAccess:
-			set.SetAll(access.Domain().Entitlements)
-		case EntitlementSetAccess:
-			set.SetAll(access.Entitlements)
-		}
-	})
-	t.EffectiveInterfaceConformanceSet().ForEach(func(it *InterfaceType) {
-		set.SetAll(it.SupportedEntitlements())
-	})
+	t.supportedEntitlementsOnce.Do(func() {
+		set = orderedmap.New[EntitlementOrderedSet](t.Members.Len())
```

Comment:
Maybe also use `:=` here like below 
```suggestion
		set := orderedmap.New[EntitlementOrderedSet](t.Members.Len())
```

and change the return type from `(set *EntitlementOrderedSet)` to `*EntitlementOrderedSet`